src/prefect/_internal/concurrency/cancellation.py

```diff
@@ -270,6 +270,8 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             # Mark as cancelled
             self.cancel(throw=False)
 
+        self._anyio_scope.__exit__(exc_type, exc_val, exc_tb)
```

Comment:
OHHHHH

Reply:
Can probably also `del self._anyio_scope` afterward too

Reply:
I didn't have time to test that out but added a TODO and will get back to it.